// kinvey-nativescript-sdk/lib/angular/endpoint.service.js
import { Inject, Injectable } from '@angular/core';
import { init, CustomEndpoint } from '../nativescript';
import { KinveyConfigToken } from './utils';
import * as i0 from "@angular/core";
export class EndpointService {
    constructor(config) {
        init(config);
    }
    endpoint(endpoint, args, options) {
        return CustomEndpoint.execute(endpoint, args, options);
    }
    execute(endpoint, args, options) {
        return this.endpoint(endpoint, args, options);
    }
}
EndpointService.ɵfac = function EndpointService_Factory(t) { return new (t || EndpointService)(i0.ɵɵinject(KinveyConfigToken)); };
EndpointService.ɵprov = i0.ɵɵdefineInjectable({ token: EndpointService, factory: EndpointService.ɵfac, providedIn: 'root' });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(EndpointService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return [{ type: undefined, decorators: [{
                type: Inject,
                args: [KinveyConfigToken]
            }] }]; }, null); })();
