// kinvey-nativescript-sdk/lib/angular/kinvey.module.js
import { NgModule } from '@angular/core';
import { KinveyConfigToken } from './utils';
import { DataStoreService } from './datastore.service';
import { EndpointService } from './endpoint.service';
import { FilesService } from './files.service';
import { PingService } from './ping.service';
import { UserService } from './user.service';
import * as i0 from "@angular/core";
export class KinveyModule {
    static init(config) {
        return {
            ngModule: KinveyModule,
            providers: [
                DataStoreService,
                EndpointService,
                FilesService,
                PingService,
                UserService,
                { provide: KinveyConfigToken, useValue: config }
            ]
        };
    }
}
KinveyModule.ɵmod = i0.ɵɵdefineNgModule({ type: KinveyModule });
KinveyModule.ɵinj = i0.ɵɵdefineInjector({ factory: function KinveyModule_Factory(t) { return new (t || KinveyModule)(); } });
/*@__PURE__*/ (function () { i0.ɵsetClassMetadata(KinveyModule, [{
        type: NgModule
    }], null, null); })();
